includes/oauth/class-mailchimp-api.php

```diff
@@ -0,0 +1,180 @@
+<?php
+/**
+ * Newspack's Mailchimp API handeling.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Mailchimp API Settings.
+ */
+class Mailchimp_API {
+	/** API Endpoint Placeholder, needs to replace <dc> by the right datacenter. */
+	const API_ENDPOINT_PLACEHOLDER = 'https://<dc>.api.mailchimp.com/3.0';
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_action( 'rest_api_init', [ __CLASS__, 'register_api_endpoints' ] );
+	}
+
+	/**
+	 * Register the endpoints.
+	 */
+	public static function register_api_endpoints() {
+		// Get Mailchimp auth status.
+		\register_rest_route(
+			NEWSPACK_API_NAMESPACE,
+			'/oauth/mailchimp',
+			[
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => [ __CLASS__, 'api_mailchimp_auth_status' ],
+				'permission_callback' => function() {
+					return current_user_can( 'manage_options' );
+				},
+			]
+		);
+
+		// Verify and save Mailchimp API Key.
+		\register_rest_route(
+			NEWSPACK_API_NAMESPACE,
+			'/oauth/mailchimp',
+			[
+				'methods'             => \WP_REST_Server::EDITABLE,
+				'callback'            => [ __CLASS__, 'api_mailchimp_save_key' ],
+				'permission_callback' => function() {
+					return current_user_can( 'manage_options' );
+				},
+				'args'                => [
+					'api_key' => [
+						'type'              => 'string',
+						'sanitize_callback' => 'sanitize_text_field',
+						'required'          => true,
+					],
+				],
+			]
+		);
+
+		// Delete Mailchimp API Key.
+		\register_rest_route(
+			NEWSPACK_API_NAMESPACE,
+			'/oauth/mailchimp',
+			[
+				'methods'             => \WP_REST_Server::DELETABLE,
+				'callback'            => [ __CLASS__, 'api_mailchimp_delete_key' ],
+				'permission_callback' => function() {
+					return current_user_can( 'manage_options' );
+				},
+			]
+		);
+	}
+
+	/**
+	 * Check whether we have an API key and its validity.
+	 *
+	 * @return WP_REST_Response
+	 */
+	public static function api_mailchimp_auth_status() {
+		// 'newspack_mailchimp_api_key' is a new option introduced to manage MC API key accross Newspack plugins.
+		// Keeping the old option for backwards compatibility.
+		$mailchimp_api_key = get_option( 'newspack_mailchimp_api_key', get_option( 'newspack_newsletters_mailchimp_api_key' ) );
```

Comment:
I'm not sure if the legacy option should be handled here. There's no backwards-compatibility concern since this is a new piece of UI (as opposed to changes in `newspack-newsletters` and `newspack-popups`). But maybe I'm overthinking it (and surely [contradicting myself](https://github.com/Automattic/newspack-plugin/pull/1168#discussion_r724295942)!)

Reply:
Probably overthinking it 😅 If we remove it from here we'll get the feeling that this is still managed from the newsletter plugin + the Mailchimp connection will raise a false negative :) let's keep it there for now.